src/core/lib/transport/static_metadata.h

```diff
@@ -521,11 +524,16 @@ typedef union {
   } named;
 } grpc_metadata_batch_callouts;
 
-#define GRPC_BATCH_INDEX_OF(slice)                                        \
-  (GRPC_IS_STATIC_METADATA_STRING((slice))                                \
-       ? static_cast<grpc_metadata_batch_callouts_index>(                 \
-             GPR_CLAMP(GRPC_STATIC_METADATA_INDEX((slice)), 0,            \
-                       static_cast<intptr_t>(GRPC_BATCH_CALLOUTS_COUNT))) \
+#define GRPC_BATCH_INDEX_OF(slice)                                            \
+  (GRPC_IS_STATIC_METADATA_STRING((slice))                                    \
```

Comment:
Can we rewrite this so it's more readable? E.g.,:

GRPC_IS_STATIC_METADATA_STRING(slice) && reinterpret_cast<grpc_core::StaticSliceRefcount*>((slice).refcount)->index <= static_cast<uint32_t>(GRPC_BATCH_CALLOUTS_COUNT) ? static_cast<grpc_metadata_batch_callouts_index>(reinterpret_cast<grpc_core::StaticSliceRefcount*>(     (slice).refcount)->index)) : GRPC_BATCH_CALLOUTS_COUNT

Reply:
Done.